sound/soc/sof/debug.c

```diff
@@ -17,6 +17,78 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+static ssize_t sof_dfsentry_write(struct file *file, const char __user *buffer,
+				  size_t count, loff_t *ppos)
+{
+	struct snd_sof_dfsentry *dfse = file->private_data;
+	struct snd_sof_dev *sdev = dfse->sdev;
+	struct sof_ipc_cmd_hdr hdr;
+	struct sof_ipc_reply reply;
+	size_t size;
+	char *string;
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG)
+	int i, err, ret;
+	unsigned long ipc_count;
+#endif
+	string = kzalloc(count, GFP_KERNEL);
+	if (!string)
+		return -ENOMEM;
+
+	size = simple_write_to_buffer(string, count, ppos, buffer, count);
+	ret = size;
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG)
+	/* check if "ipc_test" debugfs entry */
+	if (strcmp(dfse->dfsentry->d_name.name, "ipc_test"))
+		return -EINVAL;
+
+	ret = pm_runtime_get_sync(sdev->dev);
+	if (ret < 0) {
+		dev_err_ratelimited(sdev->dev,
+				    "error: enum put failed to resume %d\n",
+				    ret);
+		pm_runtime_put_noidle(sdev->dev);
+		return ret;
+	}
+
+	ret = kstrtoul(string, 10, &ipc_count);
+	if (ret < 0)
+		return ret;
+
+	/* configure test IPC */
+	hdr.cmd = SOF_IPC_GLB_TEST | SOF_IPC_TEST_IPC_FLOOD;
+	hdr.size = sizeof(hdr);
+
+	/* send test IPC's */
+	for (i = 0; i < ipc_count; i++) {
```

Comment:
@ranj063 
Will this block the write process? I think there will be long time hang after we write something

Reply:
@xiulipan yes depending on how many time you want to send the IPC's. But the dmesg will show you the progress in terms of the ipc's being sent and replies received isnt it?

Reply:
@ranj063 
What if we rewrite something into the debugfs while this one is running?

Reply:
@xiulipan are you worried that the first flood test will be affected by the second flood test? 
I tried to write while a test was in progress and didnt see anything unusual